.github/workflows/test-database.yml

```diff
@@ -0,0 +1,185 @@
+---
+name: Database testing
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - "main"
+      - "rc"
+      - "hotfix-rc"
+    paths:
+      - ".github/workflows/infrastructure-tests.yml" # This file
+      - "src/Sql/**" # SQL Server Database Changes
+      - "util/Migrator/**" # New SQL Server Migrations
+      - "util/MySqlMigrations/**" # Changes to MySQL
+      - "util/PostgresMigrations/**" # Changes to Postgres
+      - "util/SqliteMigrations/**" # Changes to Sqlite
+      - "src/Infrastructure.Dapper/**" # Changes to SQL Server Dapper Repository Layer
+      - "src/Infrastructure.EntityFramework/**" # Changes to Entity Framework Repository Layer
+      - "test/Infrastructure.IntegrationTest/**" # Any changes to the tests
+  pull_request:
+    paths:
+      - ".github/workflows/infrastructure-tests.yml" # This file
+      - "src/Sql/**" # SQL Server Database Changes
+      - "util/Migrator/**" # New SQL Server Migrations
+      - "util/MySqlMigrations/**" # Changes to MySQL
+      - "util/PostgresMigrations/**" # Changes to Postgres
+      - "util/SqliteMigrations/**" # Changes to Sqlite
+      - "src/Infrastructure.Dapper/**" # Changes to SQL Server Dapper Repository Layer
+      - "src/Infrastructure.EntityFramework/**" # Changes to Entity Framework Repository Layer
+      - "test/Infrastructure.IntegrationTest/**" # Any changes to the tests
+
+jobs:
+  test:
+    name: Run tests
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
+
+      - name: Set up .NET
+        uses: actions/setup-dotnet@3447fd6a9f9e57506b15f895c5b76d3b197dc7c2 # v3.2.0
+
+      - name: Restore tools
+        run: dotnet tool restore
+
+      - name: Docker Compose databases
+        working-directory: "dev"
+        # We could think about not using profiles and pulling images directly to cover multiple versions
+        run: |
+          cp .env.example .env
+          docker compose --profile mssql --profile postgres --profile mysql up -d
+        shell: pwsh
+
+      # I've seen the SQL Server container not be ready for commands right after starting up and just needing a bit longer to be ready
```

Comment:
🌱 Is there a way to poll it for readiness? This is more a speculative question than something that requires adjustment.